### PR TITLE
Add GetByDate

### PR DESCRIPTION
**Pull request by Evgentik**

### CODE REVIEW

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs [hunk 1]
```diff
@@ -20,6 +20,17 @@ public ScheduleRepository(IOptions<MongoSettings> options)
...
+            return GetCollectionByDatePeriod(filter, date, 1);
+        }
+
+        public Task<List<Schedule>> GetByDateAsync(Guid userId, DateTime date)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
+            return GetCollectionByDatePeriod(filter, date, 1);
+        }
+
+        private Task<List<Schedule>> GetCollectionByDatePeriod(FilterDefinition<Schedule> filter, DateTime date, int countDay)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:32]: daysNum

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs [hunk 2]
```diff
@@ -22,9 +22,15 @@ public SchedulesController(IScheduleService scheduleService)
         }
 
         [HttpGet]
-        public Task<IEnumerable<ScheduleDTO>> GetByUserPrimarySkillAsync([FromQuery, Required] Guid projectId, [FromQuery, Required] DateTime date, [FromQuery, Required] Guid primarySkillId)
+        public Task<IEnumerable<ScheduleDTO>> GetByUserPrimarySkillAsync([FromQuery, Required] Guid projectId, [FromQuery] DateTime? date, [FromQuery, Required] Guid primarySkillId)
         {
             return _scheduleService.GetByUserPrimarySkillAsync(projectId, date, primarySkillId);
         }
+
+        [HttpGet("date")]
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:30]: current_user think will be better, cause we have date in both endpoints

File: Recrutify/Recrutify.Services/Services/ScheduleService.cs [hunk 3]
```diff
@@ -16,18 +18,37 @@ public class ScheduleService : IScheduleService
...
             _mapper = mapper;
+            _userProvider = userProvider;
         }
 
-        public async Task<IEnumerable<ScheduleDTO>> GetByUserPrimarySkillAsync(Guid projectId, DateTime date, Guid primarySkillId)
+        public async Task<IEnumerable<ScheduleDTO>> GetByUserPrimarySkillAsync(Guid projectId, DateTime? date, Guid primarySkillId)
         {
+            if (!date.HasValue)
+            {
+                date = DateTime.UtcNow;
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/ScheduleService.cs:35]: DateTime.UtcNow.Date

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs [hunk 4]
```diff
@@ -20,6 +20,17 @@ public ScheduleRepository(IOptions<MongoSettings> options)
         public Task<List<Schedule>> GetByUserPrimarySkillAsync(IEnumerable<Guid> userIds, DateTime date, Guid primarySkillId)
         {
             var filter = _filterBuilder.In(u => u.UserId, userIds) & _filterBuilder.Eq(u => u.UserPrimarySkill.Id, primarySkillId);
+            return GetCollectionByDatePeriod(filter, date, 1);
+        }
+
+        public Task<List<Schedule>> GetByDateAsync(Guid userId, DateTime date)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:28]: This method will always return one document. So the one option make your private method returning FindDefinition (without ToListAsync) and in your public methods call ToListAsync and FirstOrDefaultAsync. Or in this method call await and call FirstOrDefaultAsync for your list. I prefer first option

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs (hunk 1 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:32]: +

File: Recrutify/Recrutify.Host/Controllers/SchedulesController.cs (hunk 2 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Host/Controllers/SchedulesController.cs:30]: +

File: Recrutify/Recrutify.Services/Services/ScheduleService.cs (hunk 3 above)
Comment (Evgentik) [reply Recrutify/Recrutify.Services/Services/ScheduleService.cs:35]: +

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs (hunk 4 above)
Comment (Evgentik) [reply Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:28]: +

Comment (elizavetapavliv) [reply Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:28]: FirstOrDefaultAsync

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs [hunk 5]
```diff
@@ -20,19 +20,29 @@ public ScheduleRepository(IOptions<MongoSettings> options)
...
+            return GetCollectionByDatePeriod(filter, date, 1).ToListAsync();
+        }
+
+        public Task<Schedule> GetByDatePeriodAsync(Guid userId, DateTime date, int daysNum)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
+            return GetCollectionByDatePeriod(filter, date, daysNum).FirstOrDefaultAsync();
+        }
+
+        private IFindFluent<Schedule, Schedule> GetCollectionByDatePeriod(FilterDefinition<Schedule> filter, DateTime date, int daysNum)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:32]: GetFindFluentByDate

File: Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs [hunk 6]
```diff
@@ -20,19 +20,29 @@ public ScheduleRepository(IOptions<MongoSettings> options)
...
+            return GetFindFluentByDate(filter, date, 1).ToListAsync();
+        }
+
+        public Task<Schedule> GetByDatePeriodAsync(Guid userId, DateTime date, int daysNum)
+        {
+            var filter = _filterBuilder.Eq(u => u.UserId, userId);
+            return GetFindFluentByDate(filter, date, daysNum).FirstOrDefaultAsync();
+        }
+
+        private IFindFluent<Schedule, Schedule> GetFindFluentByDate(FilterDefinition<Schedule> filter, DateTime date, int daysNum)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/ScheduleRepository.cs:32]: may be write daysNum=1 ? so you don't need pass it from 23, it's just an option to try

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**